src/repeated_game.jl

```diff
@@ -0,0 +1,339 @@
+#=
+This file contains code to build and manage repeated games
+
+It currently only has tools for solving two player repeated
+games, but could be extended to do more with some effort.
+=#
+
+"""
+This is a type for a specific type of repeated games
+
+It takes a stage game that is repeated in every period
+and all agents discount future at rate δ
+"""
+immutable RepeatedGame{N, T<:Real}
+    sg::NormalFormGame{N, T}
+    δ::Float64
+end
+
+#
+# Helper Functions (for 2 players)
+#
+
+"""
+Helper constructor that builds game from players
+"""
+RepeatedGame(p1::Player, p2::Player, δ::Float64) =
+    RepeatedGame(NormalFormGame((p1, p2)), δ)
+
+"""
+Unpacks the elements of a repeated game
+"""
+unpack(rpd::RepeatedGame) = (rpd.sg, rpd.δ)
+
+# Flow utility in terms of the players actions
+flow_u_1(rpd, a1::Int, a2::Int) = (1-rpd.δ)*rpd.sg.players[1].payoff_array[a1, a2]
+flow_u_2(rpd, a1::Int, a2::Int) = (1-rpd.δ)*rpd.sg.players[2].payoff_array[a2, a1]
+flow_u(rpd, a1::Int, a2::Int) = [flow_u_1(rpd, a1, a2), flow_u_2(rpd, a1, a2)]
+
+# Computes each players best deviation given an opponent's action
+best_dev_i(rpd, i::Int, aj::Int) = indmax(rpd.sg.players[i].payoff_array[:, aj])
+best_dev_1(rpd, a2::Int) = best_dev_i(rpd, 1, a2)
+best_dev_2(rpd, a1::Int) = best_dev_i(rpd, 2, a1)
+
+# Computes the payoff of the best deviation
+best_dev_payoff_i(rpd, i::Int, aj::Int) = (1-rpd.δ)*maximum(rpd.sg.players[i].payoff_array[:, aj])
+best_dev_payoff_1(rpd, a2::Int) = (1-rpd.δ)*maximum(rpd.sg.players[1].payoff_array[:, a2])
+best_dev_payoff_2(rpd, a1::Int) = (1-rpd.δ)*maximum(rpd.sg.players[2].payoff_array[:, a1])
+
+"""
+Creates the unit circle of continuation value points starting at origin `o`
+with radius `r`. Additionally, this function crops any points that go
+beyond the bounds permitted by the game -- For example, if the max payoff
+for player 1 is 5 then any point that awards player 1 higher than 5
+continuation utility is cropped to 5.
+"""
+function create_unitcircle_points{T<:Real}(rpd::RepeatedGame{2, T}, nH::Int,
```

Comment:
I don't see what this function is doing. Is this deriving the H-representation of script_W (the rectangle of feasible payoff profiles) given `H = unitcircle(nH)`? If that's the case, wouldn't the following work?

```jl
H = unitcircle(nH)
HT = H'

p1_extrema = extrema(g.players[1].payoff_array)
p2_extrema = extrema(g.players[2].payoff_array)
W0_vertices = gridmake([p1_extrema...], [p2_extrema...])

C = vec(maximum(W0_vertices * HT, 1))
```

* Is `Z` used in the main loop?

* This does not appear creating unit circle points; to me, a unit circle is a circle with radius one, or the circle with radius one whose center is the origin.